### PR TITLE
GOVUK-APP-630

### PR DESCRIPTION
**Pull request by raphaelolu**
Added clear all functionality to recent activity

### CODE REVIEW

File: Tests/govuk_ios/govuk_ios_unit_tests/Specs/ViewModels/RecentActivitiesViewModelTests.swift [hunk 1]
```diff
@@ -123,62 +125,222 @@ struct RecentActivitiesViewModelTests {
...
+        let activityOne = ActivityItem(context: coreData.backgroundContext)
+        activityOne.id = UUID().uuidString
+        activityOne.title = "benefits"
+        activityOne.url = "https://www.youtube.com/"
+        activityOne.date = Date.arrange("14/04/2016")
+        let activityTwo = ActivityItem(context: coreData.backgroundContext)
+        activityOne.id = UUID().uuidString
+        activityOne.title = "benefits"
+        activityOne.url = "https://www.youtube.com/"
+        activityOne.date = Date.arrange("14/10/2024")
```
Comment (TimGJD) [inline Tests/govuk_ios/govuk_ios_unit_tests/Specs/ViewModels/RecentActivitiesViewModelTests.swift:203]: Are the above supposed to be assigned to `activityTwo`?

File: Tests/govuk_ios/govuk_ios_unit_tests/Specs/ViewModels/RecentActivitiesViewModelTests.swift [hunk 2]
```diff
@@ -123,62 +125,222 @@ struct RecentActivitiesViewModelTests {
...
```
Comment (TimGJD) [inline Tests/govuk_ios/govuk_ios_unit_tests/Specs/ViewModels/RecentActivitiesViewModelTests.swift:150]: Are the above supposed to be assigned to `activityTwo`?

File: Tests/govuk_ios/govuk_ios_unit_tests/Specs/ViewModels/RecentActivitiesViewModelTests.swift [hunk 3]
```diff
@@ -123,62 +125,222 @@ struct RecentActivitiesViewModelTests {
...
+        #expect(sut.model.currentMonthActivities.count == 0)
+    }
+
+    @Test
+    func deleteActivities_deletesAllActivities() async throws {
+        let coreData = CoreDataRepository.arrange(
+            notificationCenter: .default
+        ).load()
+
+        let activityOne = ActivityItem(context: coreData.backgroundContext)
```
Comment (TimGJD) [inline Tests/govuk_ios/govuk_ios_unit_tests/Specs/ViewModels/RecentActivitiesViewModelTests.swift:141]: Looks like we have same local objects being created for different tests?
We could create these once outside the test function maybe to reduce line count?

File: Production/govuk_ios/Views/RecentlyViewed/RecentActivityView.swift [hunk 4]
```diff
@@ -2,84 +2,106 @@ import SwiftUI
...
-                            ),
-                            rows: rows,
-                            footer: nil
-                        )
-                    ]
-                )
+            if viewModel.isModelEmpty() {
+                RecentActivityErrorView()
+            } else {
+                if viewModel.model.todaysActivites.count >= 1 {
```
Comment (TimGJD) [inline Production/govuk_ios/Views/RecentlyViewed/RecentActivityView.swift:37]: We could do `if !viewModel.model.todaysActivites.isEmpty {}` here and below?

File: Tests/govuk_ios/govuk_ios_unit_tests/Specs/ViewModels/RecentActivitiesViewModelTests.swift (hunk 3 above)
Comment (raphaelolu) [reply Tests/govuk_ios/govuk_ios_unit_tests/Specs/ViewModels/RecentActivitiesViewModelTests.swift:141]: fixed

**Review by batesyuk — CHANGES_REQUESTED** [review]

**Pull request closed without merging**